src/posit/connect/resources.py

```diff
@@ -1,13 +1,22 @@
 import warnings
 
 from abc import ABC, abstractmethod
-from typing import Generic, List, Optional, TypeVar
+from typing import Any, Generic, List, Optional, TypeVar
+
+import requests
 
 
 T = TypeVar("T")
 
 
 class Resource(ABC, dict):
+    def __init__(self, session: requests.Session, url: str, **kwargs):
```

Comment:
It would muddy the source code since references will then need to be Optional aware

https://github.com/posit-dev/posit-sdk-py/pull/103/files#diff-e5a7adb6609971528ad9aa73e3641f622e37467b94f19bc21c847c9e4cb86aafR61